lib/socketry/exceptions.rb

```diff
@@ -0,0 +1,18 @@
+module Socketry
+  # Generic catch all for all Socketry errors
+  SocketError = Class.new(IOError)
```

Comment:
+1 on the first note.

Re: the `Class.new` syntax, personal preference. I find it less ugly than using a `; end`. YMMV.